packages-exp/auth-exp/src/core/credentials/phone.ts

```diff
@@ -92,7 +93,7 @@ export class PhoneAuthCredential implements externs.AuthCredential {
     return obj;
   }
 
-  static fromJSON(json: string | object): externs.AuthCredential | null {
+  static fromJSON(json: object | string): PhoneAuthCredential | null {
```

Comment:
I think the type is meant to be externs.AuthCredential:
https://firebase.google.com/docs/reference/js/firebase.auth.phoneauthcredential#static-fromjson

Reply:
this is the implementation, the public type has AuthCredential